pages/account.js

```
@@ -29,8 +29,14 @@ const Account = () => {
   const cancelReccuringDonation = async () => {
     setCancelLoading(true)
     try {
-      await fetch('/api/donations/delete')
-      router.reload()
+      const response = await fetch('/api/stripe/donations/delete', { method: 'DELETE' })
+      if (response.ok) {
+        const data = await response.json()
+        if (data.error) {
+          throw new Error(data.error)
```

this is in a try catch block, so does throwing a new error just get swallowed by line 41, or does it actually do something? can you test that?

---

Oh, yeah. The idea was to catch the error in the catch block and extract the message when showing the error on the screen. I did forget to add the code for the catch block. I only left a TODO comment.

I have fixed that now. Please confirm if this is fine, or I should approach it differently.